google-ads-examples/src/main/java/com/google/ads/googleads/examples/advancedoperations/AddDynamicPageFeed.java

```diff
@@ -0,0 +1,540 @@
+            .setOrigin(FeedOrigin.USER)
+            .build();
+
+    // Creates the operation.
+    FeedOperation operation = FeedOperation.newBuilder().setCreate(feed).build();
+
+    // Creates the feed service client.
+    try (FeedServiceClient feedServiceClient =
+        googleAdsClient.getLatestVersion().createFeedServiceClient()) {
+      // Adds the feed.
+      MutateFeedsResponse response =
+          feedServiceClient.mutateFeeds(String.valueOf(customerId), ImmutableList.of(operation));
+      String feedResourceName = response.getResults(0).getResourceName();
+      // Displays the result.
+      System.out.printf("Feed with resource name '%s' was created.%n", feedResourceName);
+
+      // Return a FeedDetails object.
+      return getFeed(googleAdsClient, customerId, feedResourceName);
+    }
+  }
+
+  /**
+   * Retrieves details about a feed.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID in which to create criterion.
+   * @return a FeedDetails object containing relevant information.
+   */
+  private FeedDetails getFeed(
+      GoogleAdsClient googleAdsClient, long customerId, String feedResourceName) {
+    // Constructs the query.
+    String query =
+        "SELECT feed.attributes FROM feed WHERE feed.resource_name = '" + feedResourceName + "'";
+
+    // Constructs the request.
+    SearchGoogleAdsRequest request =
+        SearchGoogleAdsRequest.newBuilder()
+            .setCustomerId(String.valueOf(customerId))
+            .setPageSize(PAGE_SIZE)
+            .setQuery(query)
+            .build();
+
+    // Issues the search request.
+    try (GoogleAdsServiceClient googleAdsServiceClient =
+        googleAdsClient.getLatestVersion().createGoogleAdsServiceClient()) {
+      SearchPagedResponse searchPagedResponse = googleAdsServiceClient.search(request);
+      // Gets the first result because we only need the single feed item we created previously.
+      GoogleAdsRow googleAdsRow = searchPagedResponse.getPage().getResponse().getResults(0);
+      // Gets the attributes list from the feed and create a map with keys of each attribute and
+      // values of each corresponding ID.
+      List<FeedAttribute> feedAttributeList = googleAdsRow.getFeed().getAttributesList();
+      HashMap<String, Long> feedAttributeMap = new HashMap<>();
+      for (FeedAttribute feedAttribute : feedAttributeList) {
+        feedAttributeMap.put(feedAttribute.getName().getValue(), feedAttribute.getId().getValue());
+      }
+      // Constructs a FeedDetails object so that we have a nice small
+      // wrapper around the details we care about for the next phase.
+      FeedDetails feedDetails =
+          new FeedDetails(
+              googleAdsRow.getFeed().getResourceName(),
+              feedAttributeMap.get("Page URL"),
+              feedAttributeMap.get("Label"));
+
+      return feedDetails;
+    }
+  }
+
+  /**
+   * Creates a feed mapping for a given feed.
```

Comment:
Can we refer to the guide for more information here? Feed Mapping is kind of esoteric - it's needed because Feeds can have arbitrary structure and we need to know how these map to the placeholder types, but a new user reading this wouldn't have context.

Reply:
This guide does not yet exist for GAAPI, but I think we should it one once it is created.